### PR TITLE
New lesson on using flask to create apis

### PR DESCRIPTION
**Pull request by walshbr**
Note: not ready for review and merging just yet. Using the netlify preview for soft launch.

### CODE REVIEW

**Comment by walshbr** [comment]
This should be ready to merge and publicize.

**Comment by mdlincoln** [comment]
I'll take a look!

https://deploy-preview-773--ph-dev.netlify.com/lessons/creating-apis-with-python-and-flask

**Comment by ianmilligan1** [comment]
I'll take a quick review too 👍

File: lessons/creating-apis-with-python-and-flask.md
Comment (ianmilligan1) [inline lessons/creating-apis-with-python-and-flask.md]: Pedantic: "finder on the top left of your desktop (the magnifying glass)" --> this is "spotlight," right? Finder is the file browser, and spotlight is in the upper-right corner. 😄 

Maybe change to

"In macOS, click the spotlight icon on the top right corner of your desktop (the magnifying glass) and type `terminal`."

File: lessons/creating-apis-with-python-and-flask.md
Comment (ianmilligan1) [inline lessons/creating-apis-with-python-and-flask.md]: OK I only have pedantic little things.

OS X has been frustratingly renamed to macOS. Might be worth doing a find and replace.

File: lessons/creating-apis-with-python-and-flask.md
Comment (ianmilligan1) [inline lessons/creating-apis-with-python-and-flask.md]: data-entries -> data entries

File: lessons/creating-apis-with-python-and-flask.md
Comment (ianmilligan1) [inline lessons/creating-apis-with-python-and-flask.md:154]: *sine qua non* - I'm a ignoramus who sometimes stumbles over things like this. Could we just say "The essential element"?

File: lessons/creating-apis-with-python-and-flask.md
Comment (ianmilligan1) [inline lessons/creating-apis-with-python-and-flask.md]: For Twitterbot, make sense to link to this _Programming Historian_ lesson?

https://programminghistorian.org/lessons/intro-to-twitterbots

**Review by ianmilligan1 — APPROVED** [review]
This is a fantastic lesson, and as somebody who's been hoping to learn flask I can't wait to give this a try soon.

I left a few comments here, none of which are dealbreakers - but the author could consider them before it goes live.

File: lessons/creating-apis-with-python-and-flask.md
Comment (mdlincoln) [inline lessons/creating-apis-with-python-and-flask.md]: I was surprised to see Artstor on here because I wasn't aware they had an API for accessing their library... and I actually couldn't find one on their site after paging around a bit. Is there a more specific link to an API for this? Or should it be replaced by another API service like Europeana?

**Review by mdlincoln — CHANGES_REQUESTED** [review]
I switched a few full PH urls to relative PH urls. And I had one question about Artstor...

**Comment by walshbr** [comment]
Pinging @smythp so that he can hop in and take a look at these small things. They're all pretty minor tweaks, so hopefully should be a quick review!

**Comment by mdlincoln** [comment]
Also I'd echo @ianmilligan1 this lesson reads so well and I'm legit excited to learn how to create an API rather than just consume it 😄

**Comment by smythp** [comment]
Hi Ian and Matthew,

Thanks for the review—I'm excited that the tutorial is almost ready to go up on the site. To the extent that it reads well, I can only credit Brandon's hard work and the excellent feedback from my reviewers.

- The spotlight change is spot on. I'm generally on Linux so some of the Mac OS terminology escapes me.
- Looks like there are six places I use OSX and not Mac OS. Makes sense to replace these.
- I think Artstor could be removed and replaced with Europeana (https://pro.europeana.eu/)
- Data-entries is actually data—entries, with an em dash, which reads OK if you realize it but which might be confusing. I'm fine with the change to "data entries" or to leave it as is.
- I generally avoid idiomatic language like *sine qua non* in technical writing but I usually allow myself a few touches which I think breathe some personality into the writing here and there. I'd personally leave *sine qua non* and the pretty colloquial "Whew!" I use elsewhere just for that reason. I'm fine with the proposed change, though, if it doesn't fit PH style.
- I like the Twitterbot link, and I've tried to link to other PH lessons elsewhere in the tutorial, so it fits well.

Brandon, sorry if I'm missing a way to check out the PR code and make changes myself but I'm guessing I don't have permission. Let me know if I can clarify anything or if there's a way for me to add to the commits in this PR. I just have to say that I love this review procedure—I'm going to be spoiled when I go into my next peer review process. Every journal should have continuous integration...


Best,
Patrick

**Comment by ianmilligan1** [comment]
Sounds good @smythp – and happy for you to keep in the colloquial language too. It's a fine line to walk which you're aware of so I'm personally happy to defer to you as the author!

File: lessons/creating-apis-with-python-and-flask.md
Comment (walshbr) [reply lessons/creating-apis-with-python-and-flask.md]: Resolved!

File: lessons/creating-apis-with-python-and-flask.md
Comment (walshbr) [reply lessons/creating-apis-with-python-and-flask.md]: Resolved!

File: lessons/creating-apis-with-python-and-flask.md
Comment (walshbr) [reply lessons/creating-apis-with-python-and-flask.md]: Resolved.

File: lessons/creating-apis-with-python-and-flask.md
Comment (walshbr) [reply lessons/creating-apis-with-python-and-flask.md]: Resolved!

File: lessons/creating-apis-with-python-and-flask.md
Comment (walshbr) [reply lessons/creating-apis-with-python-and-flask.md:154]: Deferring to Patrick.

**Comment by walshbr** [comment]
I wasn't sure about the permissions either @smythp. I went through and implemented the suggestions you made @ianmilligan1 and @mdlincoln - also added you as reviewers since you've been making comments on here, but let me know if you'd prefer to be left off? This should be good to go if you want to give the 👍 @mdlincoln

**Comment by mdlincoln** [comment]
@walshbr I appreciate it but I don't think I need to be named as a reviewer - my contribution isn't really akin to a peer review

**Comment by walshbr** [comment]
10/4 - took you off. Left @ianmilligan1 on there for the moment, but I can always take him off if he feels the same way. So let's wait to hear from him before merging.

**Comment by walshbr** [comment]
Removed! Should be good to go.

**Review by mdlincoln — APPROVED** [review]

**Pull request merged**
